extensions/harvester/sakubeconfig.go

```diff
@@ -0,0 +1,93 @@
+package harvester
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/clusters"
+)
+
+const (
+	HarvesterProviderClusterLabel = "labelSelector=provider.cattle.io=harvester"
+)
+
+// KubeConfigOpts is configuration need to get a kubeconfig with SA
+type KubeConfigOpts struct {
+	CSIClusterRoleName string `yaml:"csiclusterRoleName" json:"csiclusterRoleName" default:"harvesterhci.io:csi-driver"`
+	ClusterRoleName    string `yaml:"clusterRoleName" json:"clusterRoleName" default:"harvesterhci.io:cloudprovider"`
+	Namespace          string `yaml:"namespace" json:"namespace" default:"default"`
+	SaName             string `yaml:"serviceAccountName" json:"serviceAccountName" default:"rancherCluster"`
+}
+
+// GetHarvesterSAKubeconfig generates a kubeconfig from the harvester cluster with a custom SA for the
+// cluster which will use the kubeconfig. This is typically used when creating a downstream cluster for
+// harvester that uses the harvester cloud provider.
+func GetHarvesterSAKubeconfig(client *rancher.Client, clusterName string) ([]byte, error) {
+	client, err := rancher.NewClient(client.RancherConfig.AdminToken, client.Session)
+	if err != nil {
+		return nil, err
+	}
+
+	query, err := url.ParseQuery(HarvesterProviderClusterLabel)
+	if err != nil {
+		return nil, err
+	}
+
+	harvesterCluster, err := client.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(query)
```

Comment:
maybe check for the list of harvester clusters, so we don't get a panic. Empty list check

Reply:
added a check after checking err